elyra/api/elyra.yaml

```diff
@@ -1,107 +1,232 @@
-swagger: '2.0'
+openapi: 3.0.1
 info:
   title: Elyra API
   description: The API for the Elyra extensions
-  version: '1'
   contact:
-    name: Jupyter Project
-    url: 'https://jupyter.org'
-produces:
-  - application/json
-consumes:
-  - application/json
-parameters:
-  namespace:
-    name: namespace
-    required: true
-    in: path
-    description: The name of the namespace
-    type: string
-  resource:
-    name: resource
-    required: true
-    in: path
-    description: The name of the resource in a given namespace
-    type: string
+    name: Elyra
+    url: https://github.com/elyra-ai/elyra
+  version: "1"
+
 paths:
+
   /api/elyra.yaml:
     get:
-      produces:
-        - text/x-yaml
-      summary: Get API information (yaml)
       tags:
-        - api
+      - api
+      summary: Get API information (yaml)
       responses:
-        '200':
+        200:
           description: Returns a swagger specification in yaml
-  /api/elyra.json:
+          content:
+            text/x-yaml:
+              schema:
+                type: string
+        5XX:
+          description: Unexpected error.
+
+  /api/namespace:
     get:
-      summary: Get API information (json)
       tags:
-        - api
+      - namespace
+      summary: Get current namespaces
       responses:
-        '200':
-          description: Returns a swagger specification in json
-  '/api/metadata/{namespace}':
-    parameters:
-      - $ref: '#/parameters/namespace'
+        200:
+          description: Returns the list of current namespaces
+          content:
+            application/json:
+              schema:
+                type: object
+                properties:
+                  namespaces:
+                    type: array
+                    description: The name of the namespace.
+        5XX:
+          description: Unexpected error.
```

Comment:
i believe this path (`/api/namespace`) is missing `items`.
similar to how you have it defined for `/api/schema/{namespace}/{resource}` or `/api/metadata/{namespace}`. it would be something like:
```
items:
  $ref: '#/components/schemas/NamespaceResource'
```
where `NamespaceResource` would define the schema of what is returned by the endpoint.

Reply:
This is really weird. I just happened to be thinking about how we should create a namespace object (i.e., promote it from just the name that it is) so I had to respond.  Since it's just a name, I didn't implement `/api/namespace/{resource}` because there's simply nothing to return at this time.

I suspect we'll want a Namespace object.  They could already use display names and a list-valued schema property.  We'll also need a defaulting property here as well (once we talk about that) - and I'm sure we'll want other items upstream.

If you're okay, how about we leave this API as-is, pending further discussion, then add the `/api/namepace/{resource}` endpoint at that time?  The other option would be to go ahead and define the endpoint, and implement it to return an empty dict for the object (e.g., `{<namespace_name>: {}}`)

Reply:
sorry if i wasn't clear. i didn't mean to suggest to add/implement `/api/namespace/{resource}`. rather, what i was trying to say is that the `/api/namespace` returns some object. therefore, it should define the schema for that returned object (in `items`).

if you view the `elyra.yaml` with the Swaggwe UI you will notice the difference:

![image](https://user-images.githubusercontent.com/13156555/81464814-e74c2000-9192-11ea-8897-ec4e54c06496.png)



and clicking on the [INVALID button [-]](https://validator.swagger.io/validator/debug?url=https%3A%2F%2Fraw.githubusercontent.com%2Felyra-ai%2Felyra%2F0e300afe808def25d9a9cd33e059bfd86e1b7662%2Felyra%2Fapi%2Felyra.yaml) at the bottom returns:

```
"attribute paths.'/api/namespace'(get).responses.200.content.schema.items is missing"
```

let me know if that makes sense

Reply:
I see - thank you for the clarification.  I hadn't used the swagger app to check things out - but that's a great idea - and I'll make the fix.  It sounds like a test to better-enforce the return value is required as well.  Thank you!